pkg/controller/servicebindingrequest/sbrcontroller.go

```diff
@@ -113,18 +113,32 @@ func updateFunc(logger *log.Log) func(updateEvent event.UpdateEvent) bool {
 		isConfigMap := isOfKind(e.ObjectNew, "ConfigMap")
 
 		if isSecret || isConfigMap {
-			dataFieldsAreEqual, err := compareObjectFields(e.ObjectNew, e.ObjectOld, "data")
+			dataFieldsAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "data")
 			if err != nil {
 				logger.Error(err, "error comparing object fields: %s", err.Error())
 				return false
 			}
-
 			logger.Debug("Predicate evaluated", "dataFieldsAreEqual", dataFieldsAreEqual)
 			return !dataFieldsAreEqual
 		}
+		logger.Debug("updateFunc", "Predicate evaluated for Kind", e.ObjectOld.GetObjectKind)
+		specsAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "spec")
+		if err != nil {
+			logger.Error(err, "error comparing object fields: %s", err.Error())
```

Comment:
the message should indicate that there is an error when comparing `spec` part of the object.

Reply:
should we return immediately in  case of error? Right now, we continue - that might produce weird situations.

Reply:
Returning false now. Please refer the new commit